libcxx/src/new.cpp

```diff
@@ -20,7 +20,7 @@
 // in this shared library, so that they can be overridden by programs
 // that define non-weak copies of the functions.
 
-_LIBCPP_WEAK void* operator new(std::size_t size) _THROW_BAD_ALLOC {
+static void* operator_new_impl(std::size_t size) noexcept {
```

Comment:
Since the standard requires the following behaviour for the new_handler function: 

>  throw an exception of type bad_alloc or a class derived from bad_alloc;

nh() may throw an exception and cause some issues with the noexcept specification on operator_new_impl. Would you consider a change here to prevent problem?

Reply:
Yeah, I think this looks like an issue I introduced with this patch. It should be simple to fix, basically make the `_impl` function non-`noexcept` and add a test that catches that?